src/components/SourceCodeEditor.tsx

```diff
@@ -175,37 +164,37 @@ const SourceCodeEditor = (props: ISourceCodeEditor) => {
       data-testid={'sourceCode--applyButton'}
       onClick={updateModelFromTheEditor}
       tooltipProps={{ content: 'Sync your code', position: 'right' }}
-      isVisible={sourceCode !== ''}
+      isVisible={sourceCode !== '' && props.mode === CodeEditorMode.FREE_EDIT}
     />
   );
 
-  let customControls: ReactNode[] = [EditButton];
-  if (props.mode === CodeEditorMode.TWO_WAY_SYNC) {
-    customControls = [UndoButton, RedoButton, ClearButton];
-  } else if (props.mode === CodeEditorMode.FREE_EDIT && props.editable) {
-    customControls = [UndoButton, RedoButton, ClearButton, UpdateButton];
-  }
-
   return (
     <StepErrorBoundary>
+      {syncedCode !== sourceCode && (
+        <Alert
+          title="Any invalid code will be replaced after sync. If you don't want to lose your changes
+          please make a backup."
```

Comment:
What I was looking for was more a notice to let users know that they need to press the save button, not necessarily make a backup. They might not know it or even know where it is. I kind of wish we had a normal save button so it would be more obvious, but even a message would help.